src/send_future.rs

```diff
@@ -26,250 +23,353 @@ use crate::{inbox, Error, Handler};
 /// In case an actor's mailbox is bounded, [`SendFuture`] will yield `Pending` until the message is queued successfully.
 /// This allows an actor to exercise backpressure on its users.
 #[must_use = "Futures do nothing unless polled"]
-pub struct SendFuture<R, F, TResolveMarker> {
-    inner: SendFutureInner<R, F>,
-    phantom: PhantomData<TResolveMarker>,
+pub struct SendFuture<F, TState> {
+    sending: F,
+    state: TState,
```

Comment:
Is the "T" representing "Type" still relevant here, since now these can carry data?

Reply:
Probably, not. Happy to rename it just to `S` if you want!